packages/loader/container-loader/src/container.ts

```diff
@@ -1177,22 +1179,32 @@ export class Container
 	}
 
 	public serialize(): string {
-		assert(
-			this.attachState === AttachState.Detached,
-			0x0d3 /* "Should only be called in detached container" */,
-		);
+		if (this.attachmentData.state === AttachState.Attached || this.closed) {
+			throw new UsageError("Container must not be attached or closed.");
+		}
 
-		const appSummary: ISummaryTree = this.runtime.createSummary();
-		const protocolSummary = this.captureProtocolSummary();
-		const combinedSummary = combineAppAndProtocolSummary(appSummary, protocolSummary);
+		const attachingData =
```

Comment:
Are the two potential options for the state of the container at this point attaching or detached? And if it's detached, the attachingData should be undefined?

Reply:
yes. two possibilities. only an attaching has already created it base summary, so this is handling that, a detached container has no base summary, so we generate it on each call.